### PR TITLE
Fixed a few issues including unnecessary dependencies and shouldSendConfirmation bug

### PR DESCRIPTION
**Pull request by whitef0x0**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -319,12 +319,13 @@ module.exports = function(mongoose) {
   var sendConfirmationEmail = function(email, callback) {
     var mailOptions = JSON.parse(JSON.stringify(options.confirmMailOptions));
     mailOptions.to = email;
-
-    if (!callback) {
-      callback = options.confirmSendMailCallback;
-    }
-    transporter.sendMail(mailOptions, callback);
-  };
+    if(shouldSendConfirmation){
```
Comment (SaintDako) [inline index.js]: Space after `if` and before `{`. Also, change the tabs to 4 spaces in this block.

**Comment by SaintDako** [comment]
@whitef0x0 Any thoughts as to why the Travis CI build would be failing? Doesn't seem like it should be... hmm

**Comment by whitef0x0** [comment]
it's saying that shouldSendConfirmation is not defined in the logs

File: index.js [hunk 2]
```diff
@@ -302,7 +319,49 @@ module.exports = function(mongoose) {
   var sendConfirmationEmail = function(email, callback) {
     var mailOptions = JSON.parse(JSON.stringify(options.confirmMailOptions));
     mailOptions.to = email;
-    transporter.sendMail(mailOptions, callback);
+    if(shouldSendConfirmation){
```
Comment (SaintDako) [inline index.js]: This should be `options.shouldSendConfirmation`

**Comment by SaintDako** [comment]
Found it, just a small mistake. If you want to fix it along with making the `if` styling match the rest of the code, that would be cool.

**Comment by whitef0x0** [comment]
@SaintDako done

File: index.js [hunk 3]
```diff
@@ -319,9 +319,9 @@ module.exports = function(mongoose) {
   var sendConfirmationEmail = function(email, callback) {
     var mailOptions = JSON.parse(JSON.stringify(options.confirmMailOptions));
     mailOptions.to = email;
-    if(shouldSendConfirmation){
+    if (shouldSendConfirmation) {
```
Comment (SaintDako) [inline index.js]: This should also be `options.shouldSendConfirmation` 👽

**Comment by whitef0x0** [comment]
ok fixed that one too

**Comment by SaintDako** [comment]
Travis build is failing because of linting, which is fine, I'll merge this.

**Pull request merged**